File: frontend/src/api/news.ts
import { apiClient } from './client'
import type { NewsBriefResponse, NewsFeedResponse } from '../types/news'

export async function fetchMarketNews(limit = 20): Promise<NewsFeedResponse> {
  const { data } = await apiClient.get('/news/market', { params: { limit } })
  return data
}

export async function searchNews(keyword: string, limit = 10): Promise<NewsFeedResponse> {
  const { data } = await apiClient.get('/news/search', { params: { keyword, limit } })
  return data
}

export async function fetchXueqiuNews(limit = 20): Promise<NewsFeedResponse> {
  const { data } = await apiClient.get('/news/xueqiu', { params: { limit } })
  return data
}

export async function fetchNewsBrief(keyword: string, limit = 10): Promise<NewsBriefResponse> {
  const { data } = await apiClient.get('/news/brief', { params: { keyword, limit } })
  return data
}

File: backend/app/api/news.py
from __future__ import annotations

from fastapi import APIRouter, HTTPException, Query

from app.news.service import NewsProvider
from app.schemas.news import NewsBriefResponse, NewsFeedResponse

router = APIRouter(prefix="/news")
provider = NewsProvider()


@router.get("/market", response_model=NewsFeedResponse)
def get_market_news(limit: int = Query(default=20, ge=1, le=100)) -> NewsFeedResponse:
    return provider.get_market_feed(limit=limit)


@router.get("/search", response_model=NewsFeedResponse)
def search_news(keyword: str = Query(min_length=1), limit: int = Query(default=10, ge=1, le=50)) -> NewsFeedResponse:
    if not keyword.strip():
        raise HTTPException(status_code=422, detail="keyword cannot be empty")
    return provider.get_discussion_feed(keyword=keyword, limit=limit)


@router.get("/xueqiu", response_model=NewsFeedResponse)
def get_xueqiu_news(limit: int = Query(default=20, ge=1, le=100)) -> NewsFeedResponse:
    return provider.get_xueqiu_feed(limit=limit)


@router.get("/brief", response_model=NewsBriefResponse)
def get_news_brief(keyword: str = Query(min_length=1), limit: int = Query(default=10, ge=1, le=50)) -> NewsBriefResponse:
    if not keyword.strip():
        raise HTTPException(status_code=422, detail="keyword cannot be empty")
    return provider.get_brief(keyword=keyword, limit=limit)
